docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,1382 @@
+| `BootstrapConfigReady` | Mirrors the corresponding condition from the MachinePool's BootstrapConfig resource                             |
+| `InfrastructureReady`  | Mirrors the corresponding condition from the MachinePool's Infrastructure resource                              |
+| `MachinesAvailable`    | This condition surfaces detail of issues on the controlled machines, if any.                                    |
+| `ScalingUp`            | True if available replicas < desired replicas                                                                   |
+| `ScalingDown`          | True if replicas > desired replicas                                                                             |
+| `UpToDate`             | True if all the Machines controlled by this MachinePool are up to date (replicas = upToDateReplicas)            |
+| `Remediating`          | True if there is at least one Machine controlled by this MachinePool that is not passing health checks          |
+| `Deleted`              | True if MachinePool is deleted; Reason can be used to observe the cleanup progress when the resource is deleted |
+| `Paused`               | True if this MachinePool or the Cluster it belongs to are paused                                                |
+
+> To better evaluate proposed changes, below you can find the list of current MachinePool's conditions:
+> Ready, BootstrapReady, InfrastructureReady, ReplicasReady.
+
+Notes:
+- Conditions like `ScalingUp`, `ScalingDown`, `Remediating` are intended to provide visibility on the corresponding lifecycle operation.
+  e.g. If the scaling down operation is being blocked by a Machine having issues while deleting, this should surface with a reason/message in
+  the `ScalingDown` condition.
+- As of today MachinePool does not have a notion similar to MachineDeployment's MaxUnavailability.
+
+#### MachinePool Print columns
+
+| Current           | To be                  |
+|-------------------|------------------------|
+| `NAME`            | `NAME`                 |
+| `CLUSTER`         | `CLUSTER`              |
+| `DESIRED` (*)     | `PAUSED` (new) (*)     |
+| `REPLICAS`        | `DESIRED`              |
+| `PHASE` (deleted) | `CURRENT` (*)          |
+| `AGE`             | `READY`                |
+| `VERSION`         | `AVAILABLE` (new)      |
+|                   | `UP-TO-DATE` (renamed) |
+|                   | `AGE`                  |
+|                   | `VERSION`              |
+
+(*) visible only when using `kubectl get -o wide`
+
+Notes:
+- Print columns are not subject to any deprecation rule, so it is possible to iteratively improve print columns without waiting for the next API version.
+- During the implementation we are going to verify the resulting layout and eventually make final adjustments to the column list.
+
+### Changes to Cluster API contract
```

Comment:
I haven't read the details here in depth, but thank you very much for highlighting these changes in a dedicated section!

Reply:
Resolving, no additional feedback so far so I assume it is ok